src/vcpkg/vcpkgcmdarguments.cpp

```diff
@@ -388,16 +388,44 @@ namespace vcpkg
         if (const auto p = args.debug_env.get()) args.debug = *p;
 
         auto maybe_command = args.parser.extract_first_command_like_arg_lowercase();
+        args.forwardable_arguments = args.parser.get_remaining_args();
+        bool use_stderr = Util::contains(args.forwardable_arguments, "--x-json");
         if (auto command = maybe_command.get())
         {
             args.command = *command;
+            if (*command == "depend-info")
+            {
+                bool found = false;
+                for (const auto& arg : args.forwardable_arguments)
+                {
+                    if (found)
+                    {
+                        const auto value = Strings::ascii_to_lowercase(arg);
+                        use_stderr = value == "dot" || value == "mermaid" || value == "dgml";
+                        break;
+                    }
+                    if (Strings::case_insensitive_ascii_starts_with(arg, "--format"))
+                    {
+                        if (arg.length() > 9)
+                        {
+                            const auto value = Strings::ascii_to_lowercase(StringView(arg).substr(9));
+                            use_stderr = value == "dot" || value == "mermaid" || value == "dgml";
+                            break;
+                        }
+                        found = true;
+                    }
+                }
+            }
         }
 
-        args.forwardable_arguments = args.parser.get_remaining_args();
+        if (use_stderr)
+        {
+            msg::set_default_output_stream(OutputStream::StdErr);
+        }
```

Comment:
The ugly hack is here. But this now also works with `vcpkg depend-info 'pcl[core,visualization]' --forMat=dOt --debug > test.dot`